treeherder/webapp/api/performance_data.py

```diff
@@ -751,6 +751,29 @@ def _filter_out_retriggers(serialized_data: List[dict]) -> List[dict]:
         return serialized_data
 
 
+class PerformanceAlertTasks(generics.RetrieveAPIView):
+    queryset = PerformanceAlertSummary.objects.prefetch_related(
+        'alerts',
+        'alerts__series_signature',
+        'related_alerts',
+        'related_alerts__series_signature',
+    )
+    serializer_class = PerformanceAlertSummarySerializer
+    lookup_field = 'id'
+
+    def get(self, request, *args, **kwargs):
+        alert_summary = self.retrieve(request, *args, **kwargs)
```

Comment:
I'm not sure `self.retrieve` is the best approach here. What I've seen common throughout the code base is validating the query params, in this case just the `id` and then searching for the summary and checking it exists (using PerformanceAlertSummary.objects.get()). Doing this explicitly instead of using `self.retrieve` might be more consistent with the other implementations, but also easier to follow and understand. 

@gmierz What do you think about this suggestion?

Reply:
+1 to this, lets stick to how the rest of the code does this. To be clear, this is the method we should use here: https://github.com/mozilla/treeherder/blob/8fba65f76172e78fc4c54cd42aef2bea3e37ee28/treeherder/webapp/api/performance_data.py#L225

@beatrice-acasandrei I wonder if it would be a good idea to look into using the retrieve method everywhere? It seems like it could be useful to reduce complexity.

Reply:
If I write the code explicitly (using PerformanceAlertSummary.objects.get() in person), Instead of using `generics.RetrieveAPIView`, I can use `APIView`[0] with the `get` function to write more clearly.
(Naturally, the function `self.retrieve` is not used.)
Please refer to this as well.

[0]
https://www.django-rest-framework.org/api-guide/views/

Reply:
> +1 to this, lets stick to how the rest of the code does this. To be clear, this is the method we should use here:
> 
> https://github.com/mozilla/treeherder/blob/8fba65f76172e78fc4c54cd42aef2bea3e37ee28/treeherder/webapp/api/performance_data.py#L225
> 
> @beatrice-acasandrei I wonder if it would be a good idea to look into using the retrieve method everywhere? It seems like it could be useful to reduce complexity.

Probably worth looking into, but generally I think it depends from endpoint to endpoint, perhaps some checks are mandatory, I do not have all the context. Changing existing endpoints might also mean modifying a bit the frontend code since each status code is treated differently there.

Reply:
Let me ask you again about the contents to be modified based on the review :)
I made an API in the form of path variable (i.e, `performance/alertsummary-tasks/1`) with `generics.RetrieveAPIView`.
But, Do you want to change the API to query parameter form (i.e, `performance/alertsummary-tasks/?id=1`) with the below code for ?
https://github.com/mozilla/treeherder/blob/8fba65f76172e78fc4c54cd42aef2bea3e37ee28/treeherder/webapp/api/performance_data.py#L225

Reply:
Yes, you got it @junngo :) you'll also need to ensure that the alert id is valid by checking for it in the PerformanceAlertSummary.